### ISSUE TITLE
Int notempty should return true if 0

### ISSUE THREAD

**Issue by cryptdemon** [Respect/Validation#244]
The validation of `v::int()->notEmpty(0)->validate()` returns false.

This is not intuitive.  0 is an integer, therefore it's not empty.  

Intuitively should work as follows:
```php
v::int()->notEmpty(0)->validate();    //true
v::int()->notEmpty('')->validate();    //false
v::int()->notEmpty('0')->validate(0);    //true
v::int()->notEmpty(null)->validate(0);    //false
```

**Comment by henriquemoody** [comment]
This is related to #173 and #196.

**Comment by henriquemoody** [comment]
First of all, `notEmpty()` doesn't accept any value on constructor, so doesn't matter if you write `notEmpty(0)` or `notEmpty('')`.
The `int()` [accepts any integer value](https://github.com/Respect/Validation#vint), even if it's an integer string.
The `notEmpty()` rule uses PHP [empty()](http://php.net/empty) language construct, that't why you don't get the expected result.

```php
v::int()->notEmpty()->validate(); // PHP warning because there is no value for validate()
v::int()->notEmpty()->validate(null); // false because null is empty()
v::int()->notEmpty()->validate(''); // false because '' is empty()
v::int()->notEmpty()->validate('0'); // false because '0' is empty()
v::int()->notEmpty()->validate(0); // false because 0 is empty()
```

I will close this because it's a duplicated, but feel free to keep the discussion in #173

**Comment by henriquemoody** [comment]
Will be fixed on version 1.0, see [changelog](https://github.com/Respect/Validation/blob/311891b/CHANGELOG.md#100---2015-10-24) for more info about the version.

To fix this issue now all rules are mandatory, but you can use [optional()](https://github.com/Respect/Validation/blob/311891b/docs/Optional.md) rule to handle optional values, which now is `null` and an empty string by default.
